pkg/addons/seaweedfs/upgrade.go

```diff
@@ -50,3 +63,178 @@ func (s *SeaweedFS) Upgrade(
 
 	return nil
 }
+
+// needsScalingRestart checks if this upgrade requires SeaweedFS master pod restart
+// due to scaling from single replica to HA mode from versions < 2.7.3
+func (s *SeaweedFS) needsScalingRestart(ctx context.Context, kcli client.Client) bool {
+	logrus.Info("Checking if scaling fix is needed for upgrade from pre-2.7.3")
+
+	// Get the latest installation to use for getting the previous one
+	latest, err := kubeutils.GetLatestInstallation(ctx, kcli)
+	if err != nil {
+		logrus.Infof("Could not get latest installation: %v", err)
```

Comment:
Is this expected? If not return an error and fail. If this were to fail it would leave the installation in an unexpected state.